buildSrc/src/test/java/org/elasticsearch/gradle/NoticeTaskTest.java

```diff
@@ -0,0 +1,144 @@
+package org.elasticsearch.gradle;
+
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.Project;
+import org.junit.Rule;
+import org.junit.Test;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.rules.ExpectedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.commons.io.FileUtils.write;
```

Comment:
commons io is only available via a transitive dependency in `buildSrc`, it's not something we should be using, let's just stick to `Files`

Reply:
When I tried to assemble the project with just the Files functions I ran into issues where Charsets were not matching. IE UTF-8 wouldn't always be used. I've changed the function to simply ignore bad characters using the Java IO/NIO functions.